### ISSUE TITLE
Tilemap world_to_map does not respect scale of tilemap node

### ISSUE THREAD

**Issue by nikhilCad** [godotengine/godot#38798]
**Godot version:**
3.2.1 stable

**OS/device including version:**
Windows 7
**Issue description:**
The coordinates of 2D world_to_map  are messed up if you scale the tilemap node
**Steps to reproduce:**
1. Make some tilemap level
2. Scale the tilemap
3. Use world_to_map
**Minimal reproduction project:**

**Comment by groud** [comment]
I cannot test on my phone, but the world_to_map and map_to_world functions work in the nodes local space. (not the global one, the documentation is wrong there) 

Are you sur your problem is not that you suppose that the result should be in the global space?

**Comment by groud** [comment]
Also, please upload a minimal reproduction project, otherwise we will have to close the issue.

**Comment by nikhilCad** [comment]
Nope, that's not it, the problem is solved when I divide th return value of
world_to_map by the scale of tilemap.

Though this solution is very hacky.

**Comment by nikhilCad** [comment]
I am out of home now, I will try to upload minimal project tomorrow.

**Comment by groud** [comment]
> Nope, that's not it, the problem is solved when I divide th return value of
world_to_map by the scale of tilemap.

Well, sorry but that definitely looks like you are misunderstanding how the function works. With the world_to_map function you need the position to be in the nodes local space. So you have to transform your coordinates first.

Please confirm that you have your expected result by using something like:
```gdscript
$my_tilemap_node.world_to_map($my_tilemap_node.to_local(input_vector))
```

**Comment by nikhilCad** [comment]
So, as I said, I ak away from home so can't chechk. But I was searching for my issue on Google and found a issue here in this repo, where a guy advised to use world_to_map(to_local(vector)).  So in my code I wrote $map.world_to_map(to_local(vector)) . Which didn't worked because I miised the second $map inside parenthesis, as you said above.
I then divided the return value by scale and it worked, so I thought it must be a bug.

I believe I have got my answer, I am extremely soory to eat your time. Thank you!
